x/airdrop/keeper/keeper.go

```diff
@@ -0,0 +1,140 @@
+package keeper
+
+import (
+	"fmt"
+
+	errorsmod "cosmossdk.io/errors"
+	"cosmossdk.io/log"
+	"github.com/cosmos/cosmos-sdk/codec"
+	"github.com/cosmos/cosmos-sdk/store/prefix"
+	storetypes "github.com/cosmos/cosmos-sdk/store/types"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	sdkerrors "github.com/cosmos/cosmos-sdk/types/errors"
+
+	"github.com/Stride-Labs/stride/v22/x/airdrop/types"
+)
+
+type (
+	Keeper struct {
+		cdc      codec.BinaryCodec
+		storeKey storetypes.StoreKey
+		logger   log.Logger
+
+		accountKeeper types.AccountKeeper
+		bankKeeper    types.BankKeeper
+	}
+)
+
+func NewKeeper(
+	cdc codec.BinaryCodec,
+	storeKey storetypes.StoreKey,
+	logger log.Logger,
+
+	accountKeeper types.AccountKeeper,
+	bankKeeper types.BankKeeper,
+) Keeper {
+	return Keeper{
+		cdc:      cdc,
+		storeKey: storeKey,
+		logger:   logger,
+
+		accountKeeper: accountKeeper,
+		bankKeeper:    bankKeeper,
+	}
+}
+
+// Logger returns a module-specific logger.
+func (k Keeper) Logger() log.Logger {
+	return k.logger.With("module", fmt.Sprintf("x/%s", types.ModuleName))
+}
+
+func (k Keeper) Claim(ctx sdk.Context, claimer string) error {
```

Comment:
imo it's a bit nicer to split these up into separate files.

`claim.go` for Claim, ClaimAndStake, ClaimEarly
`airdrop.go` for Airdrop keepers
`allocation.go` for user allocations